File: Data Structures & Algorithms/search-2d-matrix/submission-0.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        for (int i = 0; i < matrix.length; i++) {
            int[] row = matrix[i];
            if (row[0] == target || row[row.length - 1] == target) {
                return true;
            }
            else if (row[0] < target && target < row[row.length - 1]) {
                //binary search row
                int l = 0, r = row.length;
                while (l < r) {
                    int m = (l + r) / 2;
                    if (row[m] == target) {
                        return true;
                    }
                    else if (row[m] < target) {
                        l = m + 1;
                    }
                    else {
                        r = m;
                    }
                }
                return false;
            }
        }
        return false;
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        l, r = 1, len(numbers)
        while (l < r):
            curSum = numbers[l - 1] + numbers[r - 1]
            if (curSum == target):
                return [l, r]
            elif curSum < target:
                l += 1;
            else:
                r -= 1;
        return [];